app/views/dashboard/form/publish/edit.html.erb

```diff
@@ -44,7 +44,11 @@
             </div>
             <%= render 'fields', form: form %>
 
-            <%= render 'dashboard/form/shared/action_footer', form: form, primary_action: :publish %>
+            <% if form.object.aasm.from_state == WorkVersion::STATE_DRAFT %>
+              <%= render 'dashboard/form/shared/action_footer', form: form, primary_action: :publish %>
+            <% else %>
+              <%= render 'dashboard/form/shared/action_footer', form: form, primary_action: :finish %>
+            <% end %>
```

Comment:
This is confusing to me, and I think it's a result of our `#prevalidate` method. What I think might be a better design is for `prevalidate` to: publish, validate, roll back to the original state. Then this code here can say `if form.object.published?` rather than using the from_state